static/js/settings_org.js

```diff
@@ -976,29 +973,20 @@ exports.build_page = function () {
         }
         if (icon) {
             url = '/json/realm/icon';
-            spinner = $('#realm-icon-upload-widget .upload-spinner-background');
-            upload_text =  $('#realm-icon-upload-widget .settings-page-upload-text');
-            delete_button = $('#realm-icon-upload-widget .settings-page-delete-button');
-            error_field = $("#realm-icon-upload-widget .image_file_input_error");
+            widget = '#realm-icon-upload-widget';
         } else {
             if (night) {
-                error_field = $("#realm-night-logo-upload-widget .image_file_input_error");
-                spinner = $("#realm-night-logo-upload-widget .upload-spinner-background");
-                upload_text = $('#realm-night-logo-upload-widget .settings-page-upload-text');
-                delete_button = $('#realm-night-logo-upload-widget .settings-page-delete-button');
+                widget = '#realm-night-logo-upload-widget';
             } else {
-                error_field = $("#realm-day-logo-upload-widget .image_file_input_error");
-                spinner = $("#realm-day-logo-upload-widget .upload-spinner-background");
-                upload_text = $('#realm-day-logo-upload-widget .settings-page-upload-text');
-                delete_button = $('#realm-day-logo-upload-widget .settings-page-delete-button');
+                widget = '#realm-day-logo-upload-widget';
             }
             url = '/json/realm/logo';
             form_data.append('night', JSON.stringify(night));
         }
-        spinner.expectOne();
-        upload_text.expectOne();
-        delete_button.expectOne();
-        error_field.expectOne();
+        const spinner = $(widget + " .upload-spinner-background").expectOne();
```

Comment:
I love this change! Makes for much more readable code.

We should try to use the `${}` syntax instead of string concatenation here, though.

Reply:
sure I'll change